File: java_training/PF_set_1/src/q6.java
import java.util.Scanner;

public class q6 {

	public static void main(String[] args) {
		Scanner scnr = new Scanner(System.in);
		int F = scnr.nextInt();
		scnr.close();
		if(F<0) {
			System.out.println("Invalid Input");
			System.exit(0);
		}
			
		System.out.printf("%.2f",convertToCentigrade(F));
	}
	static float convertToCentigrade(int F) {
		return (float) ((F-32)*(5.0/9.0));
	}

}

File: kotlin_training/One/src/day3/q6.kt
package day3

import java.util.*

/*
 Using TailRec compute factorial of the given number.
 Keep the number as < 50
 */
fun main() {
    println("Factorials!!\n Enter <50")
    val read = Scanner(System.`in`)
    val num = read.nextInt()
    println(if(num<50) cal_fact(num) else "Invalid Input")
}fun cal_fact(num:Int):Long{
    return if(num==1) 1 else num* cal_fact(num-1)

}